services/skus/controllers.go

```diff
@@ -1386,15 +1386,15 @@ func handleSubmitReceipt(svc *Service, valid *validator.Validate) handlers.AppHa
 		}
 
 		{
-			exists, err := svc.ExternalIDExists(ctx, extID)
-			if err != nil {
+			_, err := svc.orderRepo.GetByExternalID(ctx, svc.Datastore.RawDB(), extID)
+			if err != nil && !errors.Is(err, model.ErrOrderNotFound) {
 				l.Warn().Err(err).Msg("failed to lookup external id")
 
 				return handlers.WrapError(err, "failed to lookup external id", http.StatusInternalServerError)
 			}
 
-			if exists {
-				return handlers.WrapError(err, "receipt has already been submitted", http.StatusBadRequest)
+			if err == nil {
+				return handlers.WrapError(err, "receipt has already been submitted", http.StatusConflict)
```

Comment:
question(non-blocking): I know the error wont be serialised and nil wont show up in the response but essentially we are passing in a nil error to a func that is suppose to warp an error so it just looks unusual. 

Is there a reason we cant just return a 200 if the receipt is already linked, does the client need to distinguish between an attempt to link an already linked receipt?

Reply:
Thanks, good catch. The SDK will have to handle this differently and propagate the error up to the client so that the client can decide what to do.